lib/collection.js

```diff
@@ -1742,6 +1742,7 @@ Collection.prototype.aggregate = function(pipeline, options, callback) {
   options = Object.assign({}, options);
   // Ensure we have the right read preference inheritance
   options.readPreference = resolveReadPreference(options, { db: this.s.db, collection: this });
+  command.readPreference = options.readPreference;
```

Comment:
I think this is somewhat definitive proof that this error exists in `core` rather than `native`. We are correctly resolving the `readPreference` in the previous line, but you are able to identify that eventually the command generated in `core` is not decorated with the passed `readPreference`. Did you try to solve this there already?